benchmarks/helper.py

```diff
@@ -0,0 +1,222 @@
+parse results and generate csv/graphs.
+
+The script works on manually written TABLE (see below). We can add more commands
+in the future.
+
+One example usage is
+-> python benchmarks/helper.py --e2e --modes=training --devices=cuda --suites=torchbench --dtypes=float32 --nightly
+
+where
+    e2e - generates the commands, runs the benchmarks, parse the results (other options - commands/parse)
+    nightly - Looks at the nightly field in the training/inferece field of TABLE to choose most relevant compilers.
+    models - either training or inference
+    suites - benchmark suite
+
+In case, you just need to get the commands, you can replace e2e with commands
+-> python benchmarks/helper.py --commands --modes=training --devices=cuda --suites=torchbench --dtypes=float32 --nightly
+
+The commands are written in file run.sh.
+
+In case, you just want to parse the logs once you have run the commands manually, you can use this command
+-> python benchmarks/helper.py --parse --modes=training --devices=cuda --suites=torchbench --dtypes=float32 --nightly
+
+"""
+
+import argparse
+import itertools
+import os
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from matplotlib import rcParams
+from tabulate import tabulate
+
+rcParams.update({"figure.autolayout": True})
+plt.rc("axes", axisbelow=True)
+
+DEFAULT_OUTPUT_DIR = "benchmark_logs"
+
+
+TABLE = {
+    "suites": ["torchbench", "huggingface"],
+    "dtypes": ["float32", "float16", "amp"],
+    "devices": ["cuda", "cpu"],
+    "modes": ["inference", "training"],
+    # Dict of name to base_command. nightly is a special field to tell which compilers are relevant.
+    "training": {
+        "ts_nnc": "--training --speedup-ts --use-eval-mode --isolate",
+        "ts_nvfuser": "--training --nvfuser --speedup-ts --use-eval-mode --isolate",
+        "aot_eager": "--training --accuracy-aot-nop --generate-aot-autograd-stats --use-eval-mode --isolate",
+        "aot_nnc": "--training --accuracy-aot-ts-mincut --use-eval-mode --isolate",
+        "aot_nvfuser": "--training --nvfuser --accuracy-aot-ts-mincut --use-eval-mode --isolate",
+        "nightly": ["ts_nvfuser", "aot_nvfuser"],
+    },
+    # Dict of name to base_command. nightly is a special field to tell which compilers are relevant.
+    "inference": {
+        "ts_nnc": "-dcuda --isolate --speedup-ts",
+        "ts_nvfuser": "-dcuda --isolate -n100 --speedup-ts --nvfuser",
+        "trt": "-dcuda --isolate -n100 --speedup-trt",
+        "eager_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs",
+        "nnc_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs_ts --nvfuser",
+        "nvfuser_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs_ts",
+        "inductor_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --inductor",
+        "nightly": ["nvfuser_cudagraphs", "inductor_cudagraphs"],
+    },
+}
```

Comment:
@wconstab This is the data structure to specify the space for benchmarks.

For "training" and "inference" field, we have (compiler, cmd) pair. There is a special field - nightly (which is a bad name in the hindsight) where you can specify the top2/3 compilers which we are interested in. Thats why we chose the `ts_nvfuser` and `aot_nvfuser`